ethsigner/signer/hashicorp/src/main/java/tech/pegasys/ethsigner/signer/hashicorp/HashicorpSubCommand.java

```diff
@@ -101,16 +113,33 @@
   private final Path tlsKnownServerFile = null;
 
   private TransactionSigner createSigner() throws TransactionSignerInitializationException {
-    final HashicorpConfig config =
-        new HashicorpConfig(
-            signingKeyPath,
-            serverHost,
-            serverPort,
-            authFilePath,
-            timeout,
-            isTlsEnabled(),
-            getTlsKnownServerFile());
-    return HashicorpVaultSignerFactory.createSigner(config);
+    TlsOptions tlsOptions = null;
+    if (tlsEnabled) {
+      if (tlsKnownServerFile != null) {
+        tlsOptions =
+            new TlsOptions(Optional.of(TrustStoreType.WHITELIST), tlsKnownServerFile, null);
+      } else {
+        tlsOptions = new TlsOptions(Optional.empty(), null, null);
+      }
+    }
```

Comment:
This block could be nice have in a seperate function

Reply:
done